multicore_runtime/wsgi_config.py

```diff
@@ -0,0 +1,208 @@
+
+  Args:
+    handler: A handler (appinfo.URLMap)
+
+  Returns:
+    A modified url regex
+  """
+  if not handler.script and not handler.static_files and handler.static_dir:
+    return handler.url + '/(.*)'
+  else:
+    return handler.url
+
+
+def load_user_scripts_into_handlers(handlers):
+  """Preloads user scripts, wrapped in env_config middleware if present.
+
+  Args:
+    handlers: appinfo.handlers data as provided by get_module_config()
+
+  Returns:
+    A list of tuples suitable for configuring the dispatcher() app,
+    where the tuples are (url, script, app):
+      - url: The url pattern which matches this handler.
+      - script: The script to serve for this handler, as a string, or None.
+      - app: The fully loaded app corresponding to the script.
+  """
+  # `if x.login == appinfo.LOGIN_OPTIONAL` disables loading handlers
+  # that require login or admin status entirely. This is a temporary
+  # measure until handling of login-required handlers is implemented
+  # securely.
+  loaded_handlers = [
+      (x.url if x.script or x.static_files else static_dir_url(x),
+       x.script,
+       app_for_script(x.script) if x.script else static_app_for_handler(x))
+      for x in handlers if x.login == appinfo.LOGIN_OPTIONAL]
+  logging.info('Parsed handlers: %s',
+               [(url, script) for (url, script, _) in loaded_handlers])
+  return loaded_handlers
+
+
+def env_vars_from_env_config(env_config):
+  """Generate a dict suitable for updating os.environ to reflect app config.
+
+  This function only returns a dict and does not update os.environ directly.
+
+  Args:
+    env_config: The app configuration as generated by
+                vmconfig.BuildVmAppengineEnvConfig()
+
+  Returns:
+    A dict of strings suitable for e.g. `os.environ.update(values)`.
+  """
+
+  return {'SERVER_SOFTWARE': env_config.server_software,
+          'APPENGINE_RUNTIME': 'python27',
+          'APPLICATION_ID': '%s~%s' % (env_config.partition,
+                                       env_config.appid),
+          'INSTANCE_ID': env_config.instance,
+          'BACKEND_ID': env_config.major_version,
+          'CURRENT_MODULE_ID': env_config.module,
+          'CURRENT_VERSION_ID': '%s.%s' % (env_config.major_version,
+                                           env_config.minor_version),
+          'DEFAULT_TICKET': env_config.default_ticket}
+
+
+def user_env_vars_from_appinfo(appinfo):
+  """Generate a dict of env variables specified by the user in app.yaml.
+
+  This function only returns a dict and does not update os.environ directly.
+
+  Args:
+    appinfo: The configuration info (a parsed yaml object) as generated by
+              get_module_config()
+
+  Returns:
+    A dict of strings suitable for e.g. `os.environ.update(values)`.
+  """
+
+  return appinfo.env_variables or {}
+
+
+# Dictionary with thread-local contents.
+class ThreadLocalDict(UserDict.IterableUserDict, threading.local):
+  pass
```

Comment:
Instead of `pass`, put a docstring here. If a docstring is present, the `pass` statement will not be needed (even though the class is empty other than the docstring.)